File: iosApp/iosAppTests/core/ui/app_sheet/mock/AppSheetCommonMock.swift
//
//  AppSheetCommonMock.swift
//  iosAppTests
//
//  Created by Bradley Phillips on 5/1/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import shared

class AppSheetCommonMock: AppSheetRepresentable
{
    var contentToReturn: AppSheetContentRepresentable? = HelpSheet(
        title: "Help",
        instructions: [
            HelpSheet.Instruction(instruction: "Mock instruction")
        ],
        examples: [
            HelpSheet.Example(
                description: "Mock description..",
                tiles: [
                    HelpSheet.Tile(
                        letter: "G".utf16.first!,
                        state: HelpSheet.TileState.hidden
                    )
                ]
            )
        ],
        footer: "Mock Footer.."
    )
    var handlerToReturn: AppSheetViewHandleable? = nil
    var setContentPassedInNewContent: AppSheetContentRepresentable? = nil
    var setHandlerPassedInNewHandler: AppSheetViewHandleable? = nil
    
    func content() -> AppSheetContentRepresentable?
    {
        self.contentToReturn
    }
    
    func handler() -> AppSheetViewHandleable?
    {
        self.handlerToReturn
    }
    
    func setContent(newContent_ newContent: AppSheetContentRepresentable?)
    {
        self.setContentPassedInNewContent = newContent
    }
    
    func setHandler(newHandler_ newHandler: AppSheetViewHandleable?)
    {
        self.setHandlerPassedInNewHandler = newHandler
    }
}

File: androidApp/src/androidTest/java/com/megabreezy/breezybuilds_wordle/core/ui/app_sheet/mock/AppSheetCommonMock.kt
package com.megabreezy.breezybuilds_wordle.core.ui.app_sheet.mock

import com.megabreezy.breezybuilds_wordle.core.ui.app_sheet.AppSheetContentRepresentable
import com.megabreezy.breezybuilds_wordle.core.ui.app_sheet.AppSheetRepresentable
import com.megabreezy.breezybuilds_wordle.core.ui.app_sheet.AppSheetViewHandleable
import com.megabreezy.breezybuilds_wordle.feature.help.domain.model.HelpSheet

class AppSheetCommonMock: AppSheetRepresentable
{
    var contentToReturn: AppSheetContentRepresentable? = HelpSheet(
        title = "Help",
        instructions = listOf(
            HelpSheet.Instruction(instruction = "Mock Instruction")
        ),
        examples = listOf(
            HelpSheet.Example(
                description = "Mock description..",
                tiles = listOf(
                    HelpSheet.Tile(
                        letter = 'G',
                        state = HelpSheet.Tile.State.HIDDEN
                    )
                )
            )
        ),
        footer = "Mock Footer"
    )
    var handlerToReturn: AppSheetViewHandleable? = null

    override fun content() = contentToReturn

    override fun handler() = handlerToReturn

    override fun setContent(newContent: AppSheetContentRepresentable?)
    {
        contentToReturn = newContent
    }

    override fun setHandler(newHandler: AppSheetViewHandleable?)
    {
        handlerToReturn = newHandler
    }

}